### PR TITLE
update Rails API and AJAX lesson plans

### PR DESCRIPTION
**Pull request by amaseda**

### CODE REVIEW

File: 07-uxdi-collaboration/ajax/readme.md [hunk 1]
```diff
@@ -9,48 +9,36 @@
...
 
 ## Opening Framing (5/5)
-In the past couple weeks or so, we've been learning ALOT about server side requests and responses. Currently we know how to create applications with full CRUD on models in our database, and that's great. When we do that CRUD, however, it requires a page reload of some kind. It would be really nice if we could execute some kind of functionality on the client side of our application but still communicate with the backend without a page refresh. If only we had a client side language that was non blocking and asynchronous.
+In the past couple weeks or so, we've been learning a lot about server-side requests and responses. Currently we know how to create applications with full CRUD on models in our database, and that's great. When we do that CRUD, however, it requires a page reload of some kind. It would be really nice if had some functionality on the client side of our application but still communicate with the backend without a page refresh. If only we had a client side language that was non blocking and asynchronous...
 
 (ST-WG) Think back to the first couple weeks of class, whats the difference between synchronous and asynchronous program execution?
```
Comment (amaseda) [inline 07-uxdi-collaboration/ajax/readme.md]: Should we phrase this as why asynchronous behavior is so important here (vs. what's the difference between sync and async)?

File: 07-uxdi-collaboration/ajax/readme.md [hunk 2]
```diff
@@ -9,48 +9,36 @@
...
 
 ### T & T (10/15)
-Turn and talk to you neighbor, why might synchronus programming not be effective for the front end? Consider how http requests work within your rails application.
+Turn and talk to you neighbor, why might synchronous programming not be effective for the front end? Consider how http requests work within your rails application.
```
Comment (amaseda) [inline 07-uxdi-collaboration/ajax/readme.md]: Noticed a lot of comments in the feedback were anti-T&T. Could use this as an opportunity to either (1) create a super-guided T&T or (2) use a different exercise format. Or just leave it as is. Figured I'd bring it up here though.

Comment (amaseda) [inline 07-uxdi-collaboration/ajax/readme.md]: Another thought: do we need to ask the previous ST-WG question with this T&T right here?

File: 07-uxdi-collaboration/ajax/readme.md [hunk 3]
```diff
@@ -120,31 +104,36 @@ But Andy, i thought we were able to access that big json object we saw before? W
...
+  1. Create an input text field for City and State in the HTML.  
+  2. Have the endpoint url change dynamically based on user input to generate a div about current weather in that area.  
 
-## Break - 10/70
+## Break (10/70)
 
-## AJAX - CUD intro (5/75)
-So we've used AJAX to do an asynchronous `get` request to a third party API. But it wouldn't make sense for us to be able to do CUD functionality to that same site. They probably don't want anyone that's not a developer there to be able to update the weather however we want. That is not to say that kind of functionality doesn't exist, we just don't have access to it.
+## Intro: AJAX and CRUD (5/75)
+So we've used AJAX to do an asynchronous `GET` request to a 3rd party API. But it wouldn't make sense for us to be able to do CRUD functionality to that same site. They probably don't want anyone that's not a developer there to be able to update the weather however we want. That is not to say that kind of functionality doesn't exist, we just don't have access to it.
```
Comment (amaseda) [inline 07-uxdi-collaboration/ajax/readme.md]: Feel like there's a better way to rephrase this. Going to mull it over...

File: 07-uxdi-collaboration/ajax/readme.md (hunk 1 above)
Comment (jshawl) [inline 07-uxdi-collaboration/ajax/readme.md]: ¿porque no los dos?

File: 07-uxdi-collaboration/ajax/readme.md (hunk 2 above)
Comment (jshawl) [inline 07-uxdi-collaboration/ajax/readme.md]: recommend choosing another activity from https://docs.google.com/document/d/1M6vqcYtiBoTkoL_yYurb7YCEIXlXln0v1wi8Ol0B7XY/edit

File: 07-uxdi-collaboration/ajax/readme.md (hunk 3 above)
Comment (jshawl) [inline 07-uxdi-collaboration/ajax/readme.md]: what is the purpose of this intro? maybe a subheader could be something like "combining ajax and internal APIs". and why you would even do that

File: 07-uxdi-collaboration/ajax/readme.md [hunk 4]
```diff
@@ -120,31 +104,36 @@ But Andy, i thought we were able to access that big json object we saw before? W
...
-Once you've cloned the repo cd into it and run the commands  `$bundle install`, `$rake db:create`, `$rake db:migrate` and `$rake db:seed` in the terminal
+Once you've cloned the repo, `cd` into it and run the usual commands...
+
+```bash
+$ bundle install
+$ rake db:create db:migrate db:seed
+```
 
-We can now use `$.ajax()` to CRUD the models of our tunr app! Let's go ahead and create a new artists controller action and corresponding view: `test_ajax`
+We can now use `$.ajax()` to CRUD the models of our Tunr app! Let's go ahead and create a new Artists controller action and corresponding view: `test_ajax`
```
Comment (jshawl) [inline 07-uxdi-collaboration/ajax/readme.md]: `s/CRUD the models of/make asynchronous HTTP requests to/`

File: 07-uxdi-collaboration/ajax/readme.md [hunk 5]
```diff
@@ -174,13 +164,13 @@ We're just going to add a quick event listener to this div inside `app/assets/ja
 ```javascript
 $(document).ready(function(){
   $(".test_ajax_get").on("click", function(){
-    alert("clicked!")
+    alert("clicked!");
```
Comment (jshawl) [inline 07-uxdi-collaboration/ajax/readme.md]: I will try to use semi-colons this week

File: 07-uxdi-collaboration/ajax/readme.md [hunk 6]
```diff
@@ -246,44 +235,51 @@ $(".test_ajax_post").on("click", function(){
...
     type: 'PUT',
-    data: {artist: {photo_url: "www.google.com", name: "bobbby", nationality: "bob"}},
+    data: {
+      artist: {photo_url: "http://media.giphy.com/media/u5yMOKjUpASwU/giphy.gif",
+        name: "Robert Goulet",
+        nationality: "American"
+      }
+    },
     dataType: 'json',
     url: "http://localhost:3000/artists/6"
```
Comment (jshawl) [inline 07-uxdi-collaboration/ajax/readme.md]: remove localhost:3000, just the path

File: 07-uxdi-collaboration/ajax/readme.md (hunk 3 above)
Comment (andrewsunglaekim) [inline 07-uxdi-collaboration/ajax/readme.md]: I think it's a good place to mention alot of 3rd party API's are read only. but yeah phrasing could definitely be better.

File: 07-uxdi-collaboration/ajax/readme.md (hunk 1 above)
Comment (andrewsunglaekim) [inline 07-uxdi-collaboration/ajax/readme.md]: :+1:

File: 07-uxdi-collaboration/ajax/readme.md (hunk 2 above)
Comment (andrewsunglaekim) [inline 07-uxdi-collaboration/ajax/readme.md]: Maybe we could push the T&T to after showing google maps to start a discussion

**Comment by andrewsunglaekim** [comment]
updates looks great, i'd like to merge this in so i can add some content in. namely 2 things:

1. PKI for what they've learned up to this point and how today's lesson connects the front end to the back end for the opening framing.
2.  Also the spark notes version of the class.

**Pull request merged**